### PR TITLE
Add new table gcp_vertex_ai_model

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
select
  name,
  display_name,
  create_time,
  version_id,
  container_spec,
  deployed_models
from
  gcp_vertex_ai_model;
Warning: failed to start plugin instance 'hub.steampipe.io/plugins/turbot/prometheus@latest': Get "http://localhost:9090/api/v1/label/__name__/values?end=1711529148.242302&match%5B%5D=%7B__name__%3D~%22.%2B%22%7D&start=1711525548.242302": dial tcp 127.0.0.1:9090: connect: connection refused
+------------------------------------+------------------------------------+---------------------------+------------+--------------------------------------------------------------------------------------------------------------------------------------------------------->
| name                               | display_name                       | create_time               | version_id | container_spec                                                                                                                                          >
+------------------------------------+------------------------------------+---------------------------+------------+--------------------------------------------------------------------------------------------------------------------------------------------------------->
| microsoft_biomedclip-1711459586314 | microsoft_biomedclip-1711459586314 | 2024-03-26T18:57:25+05:30 | 1          | {"env":[{"name":"MODEL","value":"hf-hub:microsoft/BiomedCLIP-PubMedBERT_256-vit_base_patch16_224"},{"name":"TASK","value":"zero-shot-image-classificatio>
+------------------------------------+------------------------------------+---------------------------+------------+--------------------------------------------------------------------------------------------------------------------------------------------------------->
> select
  name,
  display_name,
  create_time
from
  gcp_vertex_ai_model
where
  create_time >= current_date - interval '30 days';
+------------------------------------+------------------------------------+---------------------------+
| name                               | display_name                       | create_time               |
+------------------------------------+------------------------------------+---------------------------+
| microsoft_biomedclip-1711459586314 | microsoft_biomedclip-1711459586314 | 2024-03-26T18:57:25+05:30 |
+------------------------------------+------------------------------------+---------------------------+
> select
  name,
  display_name,
  case
    when model_source_info ->> 'source_type' = '1' then 'Model Generated by AutoML'
    when model_source_info ->> 'source_type' = '2' then 'Model Imported from Custom'
    when model_source_info ->> 'source_type' = '3' then 'Model Imported from BigQuery ML'
    when model_source_info ->> 'source_type' = '4' then 'Model Saved from Model Garden'
    when model_source_info ->> 'source_type' = '5' then 'Model Saved from Genie'
    end as model_source
from
  gcp_vertex_ai_model;
+------------------------------------+------------------------------------+-------------------------------+
| name                               | display_name                       | model_source                  |
+------------------------------------+------------------------------------+-------------------------------+
| microsoft_biomedclip-1711459586314 | microsoft_biomedclip-1711459586314 | Model Saved from Model Garden |
+------------------------------------+------------------------------------+-------------------------------+

select
  name,
  display_name,
  d ->> 'endpoint' as endpoint,
  d ->> 'deployed_model_id' as deployed_model_id
from
  gcp_vertex_ai_model,
  jsonb_array_elements(deployed_models) as d
where
  d ->> 'endpoint' = 'projects/123456789/endpoints/123456789';
+------+--------------+----------+-------------------+
| name | display_name | endpoint | deployed_model_id |
+------+--------------+----------+-------------------+
+------+--------------+----------+-------------------+

select
  name,
  display_name,
  d ->> 'endpoint' as endpoint,
  d ->> 'deployed_model_id' as deployed_model_id
from
  gcp_vertex_ai_model,
  jsonb_array_elements(deployed_models) as d
where
  d ->> 'endpoint' = 'projects/979620418102/locations/us-central1/endpoints/6870247828623458304';
+------------------------------------+------------------------------------+---------------------------------------------------------------------------+---------------------+
| name                               | display_name                       | endpoint                                                                  | deployed_model_id   |
+------------------------------------+------------------------------------+---------------------------------------------------------------------------+---------------------+
| microsoft_biomedclip-1711459586314 | microsoft_biomedclip-1711459586314 | projects/979620418102/locations/us-central1/endpoints/6870247828623458304 | 6881410070668640256 |
+------------------------------------+------------------------------------+---------------------------------------------------------------------------+---------------------+
select
  name,
  display_name,
  supported_input_storage_formats
from
  gcp_vertex_ai_model
where
  supported_input_storage_formats ? 'csv';
+------------------------------------+------------------------------------+---------------------------------------------------------------------+
| name                               | display_name                       | supported_input_storage_formats                                     |
+------------------------------------+------------------------------------+---------------------------------------------------------------------+
| microsoft_biomedclip-1711459586314 | microsoft_biomedclip-1711459586314 | ["jsonl","bigquery","csv","tf-record","tf-record-gzip","file-list"] |
+------------------------------------+------------------------------------+---------------------------------------------------------------------+

select
  name,
  display_name,
  supported_input_storage_formats
from
  gcp_vertex_ai_model
where
  supported_input_storage_formats ? 'json';
+------+--------------+---------------------------------+
| name | display_name | supported_input_storage_formats |
+------+--------------+---------------------------------+
+------+--------------+---------------------------------+

select * from gcp_vertex_ai_model where name = 'turbot'
+------+------------+---------------------+---------------------+--------------+-------------+---------------------+---------------------+-------------------+--------------+--------------+-------------+-------------+------+-------------------+-----------------+-------->
| name | version_id | version_create_time | version_update_time | display_name | description | version_description | metadata_schema_uri | training_pipeline | pipeline_job | artifact_uri | create_time | update_time | etag | metadata_artifact | version_aliases | predict>
+------+------------+---------------------+---------------------+--------------+-------------+---------------------+---------------------+-------------------+--------------+--------------+-------------+-------------+------+-------------------+-----------------+-------->
+------+------------+---------------------+---------------------+--------------+-------------+---------------------+---------------------+-------------------+--------------+--------------+-------------+-------------+------+-------------------+-----------------+-------->

select * from gcp_vertex_ai_model where name = 'microsoft_biomedclip-1711459586314'
+------------------------------------+------------+---------------------------+---------------------------+------------------------------------+-------------+---------------------+---------------------+-------------------+--------------+--------------+----------------->
| name                               | version_id | version_create_time       | version_update_time       | display_name                       | description | version_description | metadata_schema_uri | training_pipeline | pipeline_job | artifact_uri | create_time     >
+------------------------------------+------------+---------------------------+---------------------------+------------------------------------+-------------+---------------------+---------------------+-------------------+--------------+--------------+----------------->
| microsoft_biomedclip-1711459586314 | 1          | 2024-03-26T18:57:25+05:30 | 2024-03-26T18:59:16+05:30 | microsoft_biomedclip-1711459586314 |             |                     |                     |                   |              |              | 2024-03-26T18:57>
+------------------------------------+------------+---------------------------+---------------------------+------------------------------------+-------------+---------------------+---------------------+-------------------+--------------+--------------+-----------------
```
</details>

### CODE REVIEW

**Comment by Priyanka-Chatterjee-2000** [comment]
No Integration test is added as no terraform resource supports it.

File: gcp/table_gcp_vertex_ai_model.go [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+			{
+				Name:        "name",
+				Type:        proto.ColumnType_STRING,
+				Description: "The resource name of the Model.",
+				Transform:   transform.FromField("Name").Transform(lastPathElement),
+			},
+			{
+				Name:        "version_id",
+				Type:        proto.ColumnType_STRING,
+				Description: "Immutable. The version ID of the model.",
```
Comment (khushboo9024) [inline gcp/table_gcp_vertex_ai_model.go:40]:
```suggestion
				Description: "The version ID of the model.",
```

File: gcp/table_gcp_vertex_ai_model.go [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+			},
+			{
+				Name:        "version_description",
+				Type:        proto.ColumnType_STRING,
+				Description: "The description of this version.",
+			},
+			{
+				Name:        "metadata_schema_uri",
+				Type:        proto.ColumnType_STRING,
+				Description: "Points to a YAML file stored on Google Cloud Storage describing additional information about the model, immutable.",
```
Comment (khushboo9024) [inline gcp/table_gcp_vertex_ai_model.go:72]:
```suggestion
				Description: "Points to a YAML file stored on Google Cloud Storage describing additional information about the model.",
```

File: gcp/table_gcp_vertex_ai_model.go [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+		Name: "projects/" + project + "/locations/" + matrixLocation + "/models/" + name,
+	}
+
+	// Call the API
+	result, err := service.Model.GetModel(ctx, req)
+	if err != nil {
+		if strings.Contains(err.Error(), "404") || strings.Contains(err.Error(), "NotFound") {
+			return nil, nil
+		}
+		logger.Error("gcp_vertex_ai_endpoint.getAIPlatformModel", "api_error", err)
```
Comment (khushboo9024) [inline gcp/table_gcp_vertex_ai_model.go:316]:
```suggestion
		logger.Error("gcp_vertex_ai_model.getAIPlatformModel", "api_error", err)
```

File: gcp/table_gcp_vertex_ai_model.go [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+				Name:        "title",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DisplayName"),
+				Description: "Title of the model, usually the same as display_name.",
+			},
+			{
+				Name:        "tags",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Labels"),
+				Description: "A map of tags for the resource, represented as a JSON object.",
```
Comment (khushboo9024) [inline gcp/table_gcp_vertex_ai_model.go:193]: @Priyanka-Chatterjee-2000  I believe we should add `standard gcp columns` in this table.

File: gcp/table_gcp_vertex_ai_model.go [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+			},
+			{
+				Name:        "original_model_info",
+				Type:        proto.ColumnType_JSON,
+				Description: "If this model is a copy of another model, this contains info about the original.",
+			},
+			// Standard columns for integration with Steampipe
+			{
+				Name:        "title",
+				Type:        proto.ColumnType_STRING,
```
Comment (khushboo9024) [inline gcp/table_gcp_vertex_ai_model.go:185]: @Priyanka-Chatterjee-2000 `akas` cloumn is missing.

**Review by khushboo9024 — APPROVED** [review]

**Pull request merged**